src/Extension/FluentExtension.php

```diff
@@ -629,6 +629,46 @@ public function onAfterWrite(): void
     {
         $this->handleClassChanged();
     }
+    
+    /**
+     * If an object is duplicated also duplicate existing localised values from original to new object.
+     */
+    public function onAfterDuplicate($original, $doWrite, $relations): void
+    {
+        $localisedTables = $this->owner->getLocalisedTables();
+
+        $suffixes = [''];
+        if ($this->owner->hasExtension(FluentVersionedExtension::class)) {
```

Comment:
FluentVersionedExtension is a subclass of FluentExtension. You could just override onAfterDuplicate on that class instead of having an inner condition here?

Reply:
simpler `$this instanceof FluentVersionedExtension` even :P